cmd/tailscale/cli/serve_v2_test.go

```diff
@@ -1029,6 +1030,98 @@ func TestCleanURLPath(t *testing.T) {
 	}
 }
 
+func TestMessageForPort(t *testing.T) {
+	e := &serveEnv{bg: true}
+
+	tests := []struct {
+		name        string
+		serveConfig *ipn.ServeConfig
+		status      *ipnstate.Status
+		dnsName     string
+		srvType     serveType
+		srvPort     uint16
+		expected    string
+	}{
+		{
+			name: "funnel-https",
+			serveConfig: &ipn.ServeConfig{
+				TCP: map[uint16]*ipn.TCPPortHandler{
+					443: {HTTPS: true},
+				},
+				Web: map[ipn.HostPort]*ipn.WebServerConfig{
+					"foo.test.ts.net:443": {
+						Handlers: map[string]*ipn.HTTPHandler{
+							"/": {Proxy: "http://127.0.0.1:3000"},
+						},
+					},
+				},
+				AllowFunnel: map[ipn.HostPort]bool{
+					"foo.test.ts.net:443": true,
+				},
+			},
+			status:  &ipnstate.Status{},
+			dnsName: "foo.test.ts.net",
+			srvType: serveTypeHTTPS,
+			srvPort: 443,
+			expected: unindent(`Available on the internet:
```

Comment:
can we move these to constants that are shared by both the test and the code? Testing strings is very fragile so at least whenever the copy changes, tests don't suddenly fail. 

If the string is being concatenated, we can abstract that into a printer function that both the test and the code can call.

Reply:
👍 done